arcgis-ios-sdk-samples/Utility network/Perform valve isolation trace/PerformValveIsolationTraceViewController.swift

```diff
@@ -104,10 +91,28 @@ class PerformValveIsolationTraceViewController: UIViewController {
         return parameters
     }
     
+    /// Load the service geodatabase and initialize the layers.
+    func loadServiceGeodatabase() {
+        serviceGeodatabase = AGSServiceGeodatabase(url: featureServiceURL)
```

Comment:
Instead of creating the geodatabase here (and needing it to be mutable and optional), could it be created where the property is declared?

```swift
let serviceGeodatabase = AGSServiceGeodatabase(url: featureServiceURL)
```

Reply:
I think the problem is `featureServiceURL` is a property, which makes the `serviceGeodatabase` cannot access it in the first place.

Reply:
Sounds like `featureServiceURL` should be `static`.

Reply:
✅  I think this is the right time to pitch the question - should we

- Create all the string/URL literals in a sample as static constants?
- Only create the strings that are required by other properties (like this URL) as static, and other strings as instance property?
- Avoid using static variables, and duplicate the strings when they are required to instantiate an object?

We may agree on a pattern so that we don't have these issues in the future.

Reply:
I don't know that there's a one-size-fits-all pattern we can follow. In this case it made sense for the URL to be static, but sometimes the URL is just used in one place, in which case it doesn't even need to be a property.